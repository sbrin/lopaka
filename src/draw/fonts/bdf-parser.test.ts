import {expect, test} from 'vitest';
import bdfFontSource from '../fonts/bdf/profont22.bdf?raw';
import parseBDF from './bdf-parser';
test('BDFParser', () => {
    const {glyphs, meta} = parseBDF(bdfFontSource);
    expect(glyphs.size).toBe(256);
    expect(meta.size).toEqual({points: 22, resolutionX: 72, resolutionY: 72});
    expect(glyphs.get(65).bytes).toMatchSnapshot();
});
