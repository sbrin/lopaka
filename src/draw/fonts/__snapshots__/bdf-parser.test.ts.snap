// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BDFParser 1`] = `
[
  12,
  0,
  30,
  0,
  30,
  0,
  51,
  0,
  51,
  0,
  97,
  128,
  97,
  128,
  192,
  192,
  255,
  192,
  255,
  192,
  192,
  192,
  192,
  192,
  192,
  192,
  192,
  192,
]
`;
